classes/query.php

```diff
@@ -1546,37 +1554,49 @@ public function process_row($row, $models, &$result)
 				// skip empty results
 				if ($isnull)
 				{
-					continue;
+					if( ! $model['relation']) 
+					{
+						continue;
+					}
+
+					$is_relation_without_records = true;
+				}
+				else
+				{
+					$pk = '['.implode('][', $pks).']';
 				}
-				$pk = '['.implode('][', $pks).']';
 			}
 
-			// do we need to check for cached objects
-			if ($this->from_cache)
+			// This part of code shouldn't run if the current model is a relation with no record(s)
+			if( ! $is_relation_without_records )
 			{
-				$obj = Model::cached_object($pk, $model['model']);
-
-				// replace the record with the cached object
-				if ($obj)
+				// do we need to check for cached objects
+				if ($this->from_cache)
 				{
-					// update the object with fields missing from it
-					$new = array();
-					foreach ($record as $column => $value)
+					$obj = Model::cached_object($pk, $model['model']);
+
+					// replace the record with the cached object
+					if ($obj)
 					{
-						if (empty($obj->{$column}))
+						// update the object with fields missing from it
+						$new = array();
+						foreach ($record as $column => $value)
 						{
-							$obj->{$column} = $value;
-							$new[$column] = $value;
+							if (empty($obj->{$column}))
```

Comment:
I know that this is existing code, but reading it again, I wonder if this goes ok. 

The idea here is that if the object in cache was created by a partial select, missing colums to the cached object will be added here. But what if `$obj->{$column}` is `int(0)` or `string('')`? I can't test it now, but I wonder if this shouldn't be an `isset()` instead of an `empty()` check...